vignettes/generic.Rmd

```diff
@@ -219,7 +219,7 @@ advs <- tribble(
 derive_vars_merged(
   adsl,
   dataset_add = advs,
-  by_vars = exprs(USUBJID),
+  by_vars = exprs(!!get_admiral_option("subject_keys")[[2]]),
```

Comment:
This looks confusing to me. We don't have any rules or convention on the number of elements in "subject_keys" or which variable is placed where. You could have `exprs(USUBJID, STUDYID)`, `exprs(STUDYID, SITEID, SUBJID)`, or `exprs(USUBJID)`. I.e., selecting a certain element is not a good idea because in general you don't know what you get.

I would either add `STUDYID` to the data and use `get_admiral_option("subject_keys")` instead of `exprs(!!get_admiral_option("subject_keys")[[2]])` or just leave it as it is.

@bms63 , @rossfarrugia , what do you think?

Reply:
add `STUDYID` to the data and use `get_admiral_option("subject_keys")` would be my preference. i don't like the `[[2]]` option either - as risky for future robustness.

Reply:
This is being used in `derive_var_trtemfl()` as well.  I flagged as concerning for me https://github.com/pharmaverse/admiral/pull/2362#issuecomment-2144118821

Reply:
Hi @ProfessorP-beep do you have time to make these final changes today?

Reply:
Yea, I should be able to get to it this afternoon / early evening.